mirai_core/updater.py

```diff
@@ -104,9 +104,10 @@ async def handshake(self):
             try:
                 await self.bot.handshake()
                 if self.use_websocket:
-                    asyncio.run_coroutine_threadsafe(
+                    await asyncio.run_coroutine_threadsafe(
```

Comment:
https://docs.python.org/3/library/asyncio-task.html
run_coroutine_threadsafe is not async function.

Reply:
Oh, I realized it returns future. so it is okay

Reply:
Uh, please cancel the merge first. It returns `concurrent.futures.Future` rather than `asyncio.Future`, while only the latter is awaitable. The code does cause an Error.
However I have no idea on the _best_ way to wait for that `concurrent.futures.Future` right now. In my mind, it can be done by throwing `future.result` into `asyncio.run_in_executor`, or use `asyncio.wrap_future()`? Which one would be better?

Reply:
Or use the old solution. Which works fine.